lib/pdf/reader/cmap.rb

```diff
@@ -86,13 +62,40 @@ def size
     # Returns an array of Integers.
     #
     def decode(c)
-      # TODO: implement the conversion
-      return c unless Integer === c
-      @map[c]
+      @map.fetch(c, [])
     end
 
     private
 
+    def process_data(data, initial_mode = :none)
+      parser = build_parser(data)
+      mode = initial_mode
```

Comment:
Setting the initial state machine mode via an argument like this is a hack for sorbet.

If I initialize mode via a constant like this, then sorbet complains that some branches of the loop are unreachable.

```ruby
mode = :none
```

I could fix the unreachable code issue/bug like this:

```
mode = T.let(:none, Symbol)
```

However, I can't use `T.let()` at runtime because sorbet isn't a dependency. By making the initial state a method parameter, I can use a method sig in the RBI file to set the type to `Symbol`.